### ISSUE TITLE
Changing navbar-brand 

### ISSUE THREAD

**Issue by bryanwillis** [twbs/bootstrap#18855]
The easiest way to fix this is by adding 100% height to the image. However, as I read in a closed issue, 100% height is not universally "desired"... But the way it works now, unless the image height is set to EXACTLY 20px it will be off center and additionally overflow beyond the navbar if it is too large unless further changes are made. And if it's smaller than 20 px it will be off center. 

The demo on the docs is using an image that is the designed to be the same height and width, but many if not most sites want to use images of varying sizes and dimensions within the navbar.

Shouldn't the `.navbar-brand` be responsive  by default, much like `.img-responsive`? This seems like it would be an obvious solution and we could even use max-height if not height. 

```css
.navbar-brand>img {
	max-height: 100%;	
        width: auto;
}
```

There are literally [thousands of questions](http://stackoverflow.com/questions/19695784/how-can-i-make-bootstrap-columns-all-the-same-height) on stackoverflow on how to get navbar brand images to play nicely with the navbar. This would suggest that the current example in the docs needs to be at least expanded upon.

While the solution I put here is almost completely reliable there is a bug in firefox that I caught a couple months ago that could cause an issue with this on a very small occasion.

While this may be a reason not to to implement these changes it might be wise to note this somewhere since many people use the 100% height method already...

It looks like the FF community is working to fix this and it might already be the case in beta releases.  An easy way to fix this is to just move the padding to the image itself. 

I haven't looked into V4 much yet so I'm not totally sure if things have changed there.

**Comment by twbs-lmvtfy** [comment]
Hi @bryanwillis!

You appear to have posted a live example (http://codepen.io/bootstrapped/pen/OMXQVo.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 92, column 7 thru column 56: Duplicate ID ```` navbar ````.
* line 149, column 7 thru column 56: Duplicate ID ```` navbar ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @bryanwillis!

You appear to have posted a live example (http://codepen.io/bootstrapped/pen/KwYGwq.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 243, column 7 thru column 57: Duplicate ID ```` navbar3 ````.
* line 294, column 7 thru column 56: Duplicate ID ```` navbar ````.
* line 330, column 7 thru column 57: Duplicate ID ```` navbar3 ````.
* line 388, column 7 thru column 56: Duplicate ID ```` navbar ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by bryanwillis** [comment]
Example of FF Bug:
http://s.codepen.io/bootstrapped/debug/OMXQVo

Example of Working Navbar Brand Images:
http://s.codepen.io/bootstrapped/debug/KwYGwq

**Comment by twbs-lmvtfy** [comment]
Hi @bryanwillis!

You appear to have posted a live example (http://codepen.io/bootstrapped/pen/KwYGwq.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 500, column 23: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by bryanwillis** [comment]
Example of FF Bug:
http://s.codepen.io/bootstrapped/debug/OMXQVo

Example of Working Navbar Brand Images:
http://s.codepen.io/bootstrapped/debug/KwYGwq

**Comment by cvrebert** [comment]
We've [declared bankruptcy](https://github.com/twbs/bootstrap/issues/18362#issuecomment-160498098) on the Bootstrap v3 navbar and aren't making any more changes to it, for stability and compatibility reasons; see https://github.com/twbs/bootstrap/issues/17598#issuecomment-160248326.

> I haven't looked into V4 much yet so I'm not totally sure if things have changed there.

v4's navbars are currently a work-in-progress and still being overhauled. See #18467.

**Comment by cvrebert** [comment]
> [...] there is a bug in firefox that I caught a couple months ago that could cause an issue with this on a very small occasion. [...] It looks like the FF community is working to fix this and it might already be the case in beta releases.

Do you have a link to the Firefox bug report handy?

**Comment by bryanwillis** [comment]
I added the bug [reports to the demo of this issue here](http://codepen.io/bootstrapped/full/OMXQVo/). The most active report is [1231059](https://bugzilla.mozilla.org/show_bug.cgi?id=1231059).

I just checked out the V4 [navbar brand docs](http://v4-alpha.getbootstrap.com/components/navbar/#brand) and noticed that there is not navbar brand example which I think is definitely good for the time being! I'll put together a demo like my previous one with different implementations of the `.navbar-header` and `.navbar-brand` in V4 when I get a minute later this week.

**Comment by twbs-lmvtfy** [comment]
Hi @bryanwillis!

You appear to have posted a live example (http://codepen.io/bootstrapped/pen/OMXQVo.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 16, column 46 thru column 49: Stray end tag ```` a ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by bryanwillis** [comment]
I added the bug [reports to the demo of this issue here](http://s.codepen.io/bootstrapped/debug/OMXQVo). The most active report is [1231059](https://bugzilla.mozilla.org/show_bug.cgi?id=1231059).

I just checked out the V4 [navbar brand docs](http://v4-alpha.getbootstrap.com/components/navbar/#brand) and noticed that there is not navbar brand example which I think is definitely good for the time being! I'll put together a demo like my previous one with different implementations of the `.navbar-header` and `.navbar-brand` in V4 when I get a minute later this week.
